From c946e73653fdd4c0455f2d45b1225a6e99ef5e54 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=D0=BD=D0=B0=D0=B1?= <nabijaczleweli@gmail.com>
Date: Sat, 9 May 2020 15:17:42 +0200
Subject: [PATCH] build: fix out-of-tree git builds

The generated Makefile used to assume that the git repository was in the
working directory, if at all present; this broke, e.g., Debian package
builds from a git repository, since the package is built with
"cd obj-$TRIPLE && ../configure"

The $(realpath) invocation subtly fixes vendoring the neomutt-test-files
repository as well, since it strips files that do not exist
(like file/missing_symlink) that make errored on previously,
since git_ver.c depended on them
---
 Makefile.autosetup | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/Makefile.autosetup b/Makefile.autosetup
index 34cc7160e..1449392b8 100644
--- a/Makefile.autosetup
+++ b/Makefile.autosetup
@@ -56,7 +56,7 @@ UNINSTALL_TARGETS=	@UNINSTALL_TARGETS@
 
 VPATH=		@VPATH@
 
-ALL_FILES!=	(cd $(SRCDIR) && git ls-files 2>/dev/null) || true
+ALL_FILES:=	$(realpath $(patsubst %,$(SRCDIR)/%,$(shell git -C $(SRCDIR) ls-files 2>/dev/null || true)))
 
 ###############################################################################
 # neomutt
@@ -568,10 +568,9 @@ $(PGPEWRAP): $(PGPEWRAPOBJS)
 
 # generated
 git_ver.c: $(ALL_FILES)
-	version=`git describe --dirty --abbrev=6 --match "20[0-9][0-9][0-9][0-9][0-9][0-9]" 2> /dev/null | \
-		sed -e 's/^[0-9]\{8\}//; s/-g\([a-z0-9]\{6\}\)/-\1/'`; \
-	echo 'const char *GitVer = "'$$version'";' > $@.tmp; \
-	cmp -s $@.tmp $@ || mv $@.tmp $@; \
+	echo 'const char *GitVer = "$(shell git -C $(SRCDIR) describe --dirty --abbrev=6 --match "20[0-9][0-9][0-9][0-9][0-9][0-9]" 2>/dev/null | \
+		sed -e 's/^[0-9]\{8\}//; s/-g\([a-z0-9]\{6\}\)/-\1/')";' > $@.tmp
+	cmp -s $@.tmp $@ || mv $@.tmp $@
 	rm -f $@.tmp
 
 hcache/hcversion.h:	$(SRCDIR)/address/address.h $(SRCDIR)/email/body.h \
-- 
2.20.1

